develop/app/Anser/Services/v2/ProductService.php

```diff
@@ -197,4 +197,112 @@ function (
             );
         return $action;
     }
+
+    /**
+     * Add inventory
+     *
+     * @param integer|null $p_key
+     * @param integer|null $addAmount
+     * @return ActionInterface
+     */
+    public function addInventory(?int $p_key = null, ?int $addAmount = null): ActionInterface
+    {
+        $action = $this->getAction("POST", "/api/v2/inventory/addInventory")
+            ->addOption("json", [
+                "p_key"     => $p_key,
+                "addAmount" => $addAmount,
+            ])
+           ->doneHandler(
+               function (
+                   ResponseInterface $response,
+                   Action $action
+               ) {
+                   $resBody = $response->getBody()->getContents();
+                   $data    = json_decode($resBody, true);
+                   $action->setMeaningData($data["status"]);
+               }
+           )
+            ->failHandler(
+                function (
+                    ActionException $e
+                ) {
+                    log_message("critical", $e->getMessage());
+                    $e->getAction()->setMeaningData(["message" => $e->getMessage()]);
+                }
+            );
+        return $action;
+    }
+
+    /**
+     * Reduce inventory
+     *
+     * @param integer|null $p_key
+     * @param integer|null $reduceAmount
+     * @return ActionInterface
+     */
+    public function reduceInventory(?int $p_key = null, ?int $reduceAmount = null): ActionInterface
```

Comment:
Ditto.
If you define the parameter the initial value, you need to have a condition to judge the parameter whether is null.